src/StreamingBodyParser/RawBodyParser.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+use React\Promise\ExtendedPromiseInterface;
+
+class RawBodyParser implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var ExtendedPromiseInterface
+     */
+    private $promise;
+
+    /**
+     * @param Request $request
+     * @return ParserInterface
+     */
+    public static function create(Request $request)
+    {
+        return new static($request);
+    }
+
+    /**
+     * @param Request $request
+     */
+    private function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->promise = ContentLengthBufferedSink::createPromise(
+            $request,
+            $headers['content-length']
```

Comment:
What happens if there is no (or an invalid)`content-length` header? I know there is a check in the `Factory`, but since this class isn't marked as internal, i think there should be an additional check here or in `create`.